azurerm/resource_arm_servicebus_queue_authorization_rule.go

```diff
@@ -0,0 +1,157 @@
+		Read:   resourceArmServiceBusQueueAuthorizationRuleRead,
+		Update: resourceArmServiceBusQueueAuthorizationRuleCreateUpdate,
+		Delete: resourceArmServiceBusQueueAuthorizationRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: azure.ServiceBusAuthorizationRuleSchemaFrom(map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateServiceBusAuthorizationRuleName(),
+			},
+
+			"namespace_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateServiceBusNamespaceName(),
+			},
+
+			"queue_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateServiceBusQueueName(),
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+		}),
+
+		CustomizeDiff: azure.ServiceBusAuthorizationRuleCustomizeDiff,
+	}
+}
+
+func resourceArmServiceBusQueueAuthorizationRuleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusQueuesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM ServiceBus Queue Authorization Rule creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	namespaceName := d.Get("namespace_name").(string)
+	queueName := d.Get("queue_name").(string)
+
+	parameters := servicebus.SBAuthorizationRule{
+		Name: &name,
+		SBAuthorizationRuleProperties: &servicebus.SBAuthorizationRuleProperties{
+			Rights: azure.ExpandServiceBusAuthorizationRuleRights(d),
+		},
+	}
+
+	_, err := client.CreateOrUpdateAuthorizationRule(ctx, resGroup, namespaceName, queueName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.GetAuthorizationRule(ctx, resGroup, namespaceName, queueName, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read ServiceBus Namespace Queue Rule %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmServiceBusQueueAuthorizationRuleRead(d, meta)
+}
+
+func resourceArmServiceBusQueueAuthorizationRuleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusQueuesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	namespaceName := id.Path["namespaces"]
+	name := id.Path["authorizationRules"]
+	queueName := id.Path["queues"]
+
+	resp, err := client.GetAuthorizationRule(ctx, resGroup, namespaceName, queueName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure ServiceBus Queue Authorization Rule %s: %+v", name, err)
```

Comment:
can we raise the other info here? IMO the names are likely to be used across resources